test/machines/machines_test.go

```diff
@@ -201,6 +201,33 @@ var _ = framework.SigKubeDescribe("Machines", func() {
 			f.DeleteMachineAndWait(testMachine, acw)
 		})
 
+		It("Can create EBS volumes", func() {
+			testMachineProviderSpec, err := utils.TestingMachineProviderSpecWithEBS(awsCredSecret.Name, cluster.Name)
+			Expect(err).NotTo(HaveOccurred())
+			testMachine := manifests.TestingMachine(cluster.Name, cluster.Namespace, testMachineProviderSpec)
+			f.CreateMachineAndWait(testMachine, acw)
+			machinesToDelete.AddMachine(testMachine, f, acw)
+
+			volumes, err := acw.GetVolumes(testMachine)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Checking EBS volume mounts", func() {
+				devs := make([]string, 0, len(volumes))
+				for dev := range volumes {
+					devs = append(devs, dev)
+				}
+				Expect(devs).To(Equal([]string{"/dev/sda1", "/dev/sdb"}))
```

Comment:
why is "/dev/sda1", "/dev/sdb" the expectation?

Reply:
That's what the instances were having. Removed that, only the type and size is relevant as you stated.

Reply:
The deviceName is relevant as well. If "That's what the instances were having" we need to understand why and validate it is consistent with the expected behaviour. 
Because we currently only the root device volume will be configured, see https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/actuators/machine/instances.go#L188 this test should validate that the volume parameters only apply to that one as expected.
As a plus you could add support for the value to be set by the incoming api field and cover the case as well

Reply:
The AMI has more than one EBS volume defined.
The test checks now that the root one get's properly configured.